### ISSUE TITLE
Office click to run issue after BIS-F seal 

### ISSUE THREAD

**Issue by matthias-schlimm** [EUCweb/BIS-F#56]
Starting to see an issue in recent Office click-to-run installations when sealing with BIS-F. After publishing the images, Office apps won't launch and event viewer shows "application was unable to start correctly (0xc0000142). Click OK to close the application". 
[BIS-F.zip](https://github.com/EUCweb/BIS-F/files/2327626/BIS-F.zip)

**Comment by matthias-schlimm** [comment]
can you run a additional test with BIS-F but disable  Citrix optimizer inside the BIS-F Policy, i will check if the issue comes from BIS-F or Citrix Optimizer

**Comment by brettkmolitor** [comment]
[bis-fnooptimizer.zip](https://github.com/EUCweb/BIS-F/files/2328978/bis-fnooptimizer.zip)

Okay - test was done without Citrix Optimizer and issue remained. I attached logs here.

**Comment by matthias-schlimm** [comment]
Inside of your BIS-F installation, please replace 96_PrepBISF_Rearm.ps1
and 10_PersBISF_OfficeKMS.ps1 with, take a backup of the old files before replace
https://github.com/EUCweb/BIS-F/blob/6.2.0_Bugs_Bunny/Framework/SubCall/Preparation/96_PrepBISF_Rearm.ps1

https://github.com/EUCweb/BIS-F/blob/6.2.0_Bugs_Bunny/Framework/SubCall/Personalization/10_PersBISF_OfficeKMS.ps1

**Comment by matthias-schlimm** [comment]
out of the replacement of the 2 files above, please check if the Windows Update Service is running on the PVS Target Device, if not start the service and test if Office can be executed without any issue.

**Comment by trhymers** [comment]
I have checked windows update service and it is running.  Office works just fine after the online repair.  I replaced the two files, and ran the script again.  Office broke again. I included the most recent log files.

[Pers_BIS_2016PVSGOLD_GEN_20180828-150813.zip](https://github.com/EUCweb/BIS-F/files/2329724/Pers_BIS_2016PVSGOLD_GEN_20180828-150813.zip)

**Comment by matthias-schlimm** [comment]
ok, currently not a Idea where it come from. Do you deploy office, as describes here https://msandbu.wordpress.com/2016/02/04/office365-on-terminal-server-done-right/ ?

**Comment by trhymers** [comment]
I will dig into that. We call a bat file that reads an xml.  here is the contents of the xml.  and it also runs a reg setting i will paste below the xml contents. 

<Configuration>
    <Add OfficeClientEdition="32" Channel="Current" SourcePath="\\vs-mpls2\DFSMapF\APPS\O365\" OfficeMgmtCOM="TRUE">
        <Product ID="O365ProPlusRetail">
            <Language ID="en-us"/>
            <ExcludeApp ID="Groove"/>
            <ExcludeApp ID="OneDrive"/>
        </Product>
    </Add>
    <Updates Enabled="TRUE" Channel="Current"/>
    <Display Level="None" AcceptEULA="TRUE"/>
    <Property Name="AUTOACTIVATE" Value="1"/>
    <Property Name="FORCEAPPSHUTDOWN" Value="FALSE"/>
    <Property Name="SharedComputerLicensing" Value="1"/>
    <Property Name="PinIconsToTaskbar" Value="FALSE"/>
	</Configuration>



Windows Registry Editor Version 5.00

[HKEY_CURRENT_USER\Software\Microsoft\Office\16.0\Common\Internet]
"OnlineStorage"=dword:00000003

[HKEY_CURRENT_USER\Software\Microsoft\Office\16.0\Common\General]
"sharedtemplates"=hex(2):48,00,3a,00,5c,00,46,00,6f,00,72,00,6d,00,73,00,2d,00,\
  53,00,74,00,64,00,73,00,5c,00,54,00,65,00,6d,00,70,00,6c,00,61,00,74,00,65,\
  00,73,00,5c,00,57,00,6f,00,72,00,64,00,00,00

[HKEY_CURRENT_USER\Software\IM Providers]
"DefaultIMApp"="Lync"

Madison: 608-298-xxxx (5400 – 5439)

**Comment by trhymers** [comment]
looks like contents of xml where yanked.  here is the file.

[SRF-RDS.zip](https://github.com/EUCweb/BIS-F/files/2329864/SRF-RDS.zip)

**Comment by matthias-schlimm** [comment]
looks good so far

**Comment by matthias-schlimm** [comment]
Please check the following on the PVS Server:
1. Select the vDisk in the Console, then right-click and select File Properties. The vDisk File Properties dialog appears.
2. Click the Microsoft Volume Licensing tab and send me the selected Licencing Mode

**Comment by matthias-schlimm** [comment]
check out https://support.office.com/en-us/article/error-0xc0000142-when-you-start-an-office-2016-application-0249adb1-06d7-41a3-b2df-da5ff57fab37?ui=en-US&rs=en-US&ad=US

do you also have the latest windows updates installed ?

**Comment by trhymers** [comment]
KMS is the licensing option. I am checking out the link you sent.  And latest updates should be applied.
But to mention again, Office works till i run BIS-f, then it breaks.  I can then run an office online repair and it works again.  Then can run the script to break it again, and continue that cycle as long as I want.

**Comment by matthias-schlimm** [comment]
Can you set it back to NONE

**Comment by trhymers** [comment]
I can, and then try to run the script again?

**Comment by trhymers** [comment]
and to mention after i run the script, the shortcuts to the offices apps are removed.

**Comment by matthias-schlimm** [comment]
Yes, please. I think it helps if you build the vDisk from scratch only 

Second test: remove 2 BIS-F office scripts completely from the bis-f installation and run the script again. Works on a fresh new vDisk only.

If the problem still exists, I think it will not work if you change something like above on the old vDisk

**Comment by matthias-schlimm** [comment]
Do you have the cleanup Startmenue configured in the BIS-F policy ?

**Comment by trhymers** [comment]
where do i remove the office scripts?  thx.  i found the cleanup starmenu, so that answers why the disappear.

**Comment by EUCweb** [comment]
please remove:

C:\Program Files (x86)\Base Image Script Framework (BIS-F)\Framework\SubCall\Preparation\96_PrepBISF_Rearm.ps1

C:\Program Files (x86)\Base Image Script Framework (BIS-F)\Framework\SubCall\Personalization\10_PersBISF_OfficeKMS.ps1

**Comment by trhymers** [comment]
ok, i kind of did this a little different and only had it run the 97_PrepBISF_PRE_BaseImage script, and it breaks office.  so i will try what you said and see what that does.

**Comment by trhymers** [comment]
I repaired office and confirmed it work.  Ran the script without those 2 being processed.  It broke again.

**Comment by EUCweb** [comment]
Does it broke immediately after you run the script or after the reboot ?

**Comment by trhymers** [comment]
immediately.

**Comment by EUCweb** [comment]
Ok, i will check the logs again and come back to you

**Comment by matthias-schlimm** [comment]
In the BIS-F GPO > Microsoft , please disable the .net optimization first and run the script again 

2. please disable the Ghost devices to remove in the BIS-F GPO as well

**Comment by trhymers** [comment]
![image](https://user-images.githubusercontent.com/42785091/44878448-87d30880-ac6c-11e8-947b-a2ba3a013f7f.png)

i disabled .net, didnt work.  which Ghost device should i disable, all 3?

**Comment by matthias-schlimm** [comment]
Yes, please disable all 3 of them

**Comment by trhymers** [comment]
i disabled them also and tried again. still breaks.

**Comment by matthias-schlimm** [comment]
Ok, please remove from the PrepBISF folder the complete script 97_PrepBISF.... 

Try and error

**Comment by trhymers** [comment]
i removed that script, and office products work.

**Comment by matthias-schlimm** [comment]
Couple of optimization running there and also deleting temp files, we are going into this next week

**Comment by vMathias79** [comment]
I think I can contribute the solution to this, because I have sort of run into this too. In my case it wasn't directly BIS-F itself, but the same approach, just executed through Ivanti EM instead of BIS-F.

The problem lies in the deletion of the AllUsersStartmenu. As the office clicktorun installer utilizes parts of App-V/App-X technology what they basically do is to use their virtualization technology from App-V to overlay some folders onto physical ones.

If you remove the program C:\ProgramData\Microsoft\Windows\Start Menu\Programs every office clicktorun component will fail to start with the outlined error.

Furthermore you will se an error entry in the eventlog which says:

Process XXXX failed to start due to Virtual Filesystem subsystem failure. Package ID {9ac08e99-230b-47e8-9721-4577b7f124ea}. Version ID {1a8308c7-90d1-4200-b16e-646f163a08e8}. Error: 0x8F10252A-0x2
Where the very last part of the error "0x2" is in App-V Error id the information: file not found.

So long story short: If you delete the allusers startmenu you have to at least to recreate the folder "Programs" inside C:\ProgramData\Microsoft\Windows\Start Menu to make an office c2r installation working.

Hope this clarifies a bit how to work around the issue. Maybe BIS-F could also extend the logic to recreate the "Programs" folder again, after deletion of the allusers startmenu...

**Comment by trhymers** [comment]
I had read some things like that.  I will try and block that part out and see what happens.

**Comment by trhymers** [comment]
so I remed out the following and ran the script.  Office products worked. 

	#$Dir_AllUsersStartMenu = "C:\ProgramData\Microsoft\Windows\Start Menu\*"
and
 #$PrepCommands += [pscustomobject]@{
			#Order="$ordercnt"; 
			#Enabled="$true"; 
			#showmessage="Y"; 
			#CLI="LIC_BISF_CLI_SM"; 
			#TestPath =""; 
			#Description="Delete AllUsers Start Menu $Dir_AllUsersStartMenu ?";                                                     
			#Command="Remove-Item '$Dir_AllUsersStartMenu' -Recurse -Force -ErrorAction SilentlyContinue"};
			#$ordercnt += 1

**Comment by matthias-schlimm** [comment]
This can be configured through the BIS-F ADMX to not delete the AllUsersStartMenu

**Comment by trhymers** [comment]
ok I will check that out.

**Comment by trhymers** [comment]
so I disabled it in policy, and everything still works.  So I guess I am left with not being able to delete the AllUsersStartMenu if I want office products to continue to work.

**Comment by matthias-schlimm** [comment]
with the current BIS-F release not possible. As @MathiasSVA has described BIS-F must re-create the "Programs" Folder after deletion to get Office 365 C2R working.

**Comment by trhymers** [comment]
OK. That is going to be beyond me at this time.  So I guess I will just not delete the start menu for now.

**Comment by matthias-schlimm** [comment]
creating Fix, take a Copy and test (line 813 function Create-AllusersStartmenuPrograms)https://github.com/EUCweb/BIS-F/blob/develop/Framework/SubCall/Preparation/97_PrepBISF_PRE_BaseImage.ps1

**Comment by trhymers** [comment]
I enabled the start menu policy.  I copied the contents of the script into 97_PrepBISF_PRE_BaseImage.ps1 and executed the initial script. It still broke office.

**Comment by trhymers** [comment]
[Doc1.docx](https://github.com/EUCweb/BIS-F/files/2501955/Doc1.docx)

**Comment by matthias-schlimm** [comment]
Office can‘t start after the sealing or did you mean, BIS-F did not detect the Office installation ?

**Comment by trhymers** [comment]
Office apps will not start.

**Comment by matthias-schlimm** [comment]
Why run‘s the office check after the final step ?

**Comment by matthias-schlimm** [comment]
But you tested it manually, if you create the programs folder. Right ?

**Comment by trhymers** [comment]
i have never tried to create the folder manually.  not sure which folder i need to created.

**Comment by trhymers** [comment]
![image](https://user-images.githubusercontent.com/42785091/47308438-38b7ad00-d5f7-11e8-88b8-a398f64e17ea.png)

**Comment by matthias-schlimm** [comment]
as @MathiasSVA descibribes above

--snip--I think I can contribute the solution to this, because I have sort of run into this too. In my case it wasn't directly BIS-F itself, but the same approach, just executed through Ivanti EM instead of BIS-F.
The problem lies in the deletion of the AllUsersStartmenu. As the office clicktorun installer utilizes parts of App-V/App-X technology what they basically do is to use their virtualization technology from App-V to overlay some folders onto physical ones.
If you remove the program C:\ProgramData\Microsoft\Windows\Start Menu\Programs every office clicktorun component will fail to start with the outlined error.
Furthermore you will se an error entry in the eventlog which says:
Process XXXX failed to start due to Virtual Filesystem subsystem failure. Package ID {9ac08e99-230b-47e8-9721-4577b7f124ea}. Version ID {1a8308c7-90d1-4200-b16e-646f163a08e8}. Error: 0x8F10252A-0x2
Where the very last part of the error "0x2" is in App-V Error id the information: file not found.
So long story short: If you delete the allusers startmenu you have to at least to recreate the folder "Programs" inside C:\ProgramData\Microsoft\Windows\Start Menu to make an office c2r installation working.
Hope this clarifies a bit how to work around the issue. Maybe BIS-F could also extend the logic to recreate the "Programs" folder again, after deletion of the allusers startmenu…
--snap--

**Comment by matthias-schlimm** [comment]
https://github.com/EUCweb/BIS-F/issues/56#issuecomment-419134012

**Comment by trhymers** [comment]
sorry, i went back and read that also.  i previously posted a screenshot of that folder.  there is a programs folder there.

**Comment by matthias-schlimm** [comment]
can you check the NTFS rights if the Programs Folder has the same rights. Otherwise replace the NTFS manually and check of office works

**Comment by trhymers** [comment]
i checked permissions against a working server and they appear to be the same.

**Comment by matthias-schlimm** [comment]
and if you disbale the "Remove AllusersStartmenu" in ADMX, it should work, i must check it in my lab, but time only for this on weekend

**Comment by trhymers** [comment]
i disabled that option in ADMX and it works as expected.

**Comment by matthias-schlimm** [comment]
can you get a copy of the content from the Programs folder and if you enable the policy and bis-f is run, please copy the content back to see if office works. 


If it works, please remove folder step-by-step to see, which folder of it affect the office installation

**Comment by trhymers** [comment]
ok i will give it a try.


-------- Original Message --------
Subject: Re: [EUCweb/BIS-F] Office click to run issue after BIS-F seal 
(#56)

**Comment by trhymers** [comment]
it seems to be working now.  i copied contents of that script file to older ps1 file.  made sure start menu and citrix optimization were enabled in GP.  ran gpupdate /force and rebooted.  then ran the initial launch script for BIS as an administrator and office worked fine.  did it on my 2 other images and everything worked fine also.  so i think all is well now.

**Comment by matthias-schlimm** [comment]
ok, thx for your reply. so this issue can be closed now ?

**Comment by trhymers** [comment]
sure.  that should be fine.  thx.


-------- Original Message --------
Subject: Re: [EUCweb/BIS-F] Office click to run issue after BIS-F seal 
(#56)

**Comment by mschaepers** [comment]
FYI
this issue happened to me with Office365 Pro Plus after running BIS-F in the latest version (v6.1.0 build 01.101)
even reinstalling office365 was failing, recreating "C:\ProgramData\Microsoft\Windows\Start Menu\Programs" made the trick and office is able to start again.

**Comment by EUCweb** [comment]
This is fixed in 102 or above, checkout https://eucweb.com/download-bis-f

**Comment by mschaepers** [comment]
Oh, I was downloading BIS-F by the login consultants download Formular, assuming that that one provides the latest version.

Thanks!

________________________________

Von: EUCweb.com <notifications@github.com>
Datum: 24. April 2019 um 16:46:05 MESZ
An: EUCweb/BIS-F <BIS-F@noreply.github.com>
Cc: Mauricio Schäpers <m.schaepers@vsqloud.de>, Comment <comment@noreply.github.com>
Betreff: Re: [EUCweb/BIS-F] Office click to run issue after BIS-F seal (#56)


This is fixed in 102 or above, checkout https://eucweb.com/download-bis-f
